rust/geoarrow-array/src/datatypes.rs

```diff
@@ -61,17 +62,7 @@ pub enum NativeType {
 
     /// Represents a mixed geometry array of unknown types or dimensions
     Geometry(GeometryType),
-}
-
-impl From<NativeType> for DataType {
-    fn from(value: NativeType) -> Self {
-        value.to_data_type()
-    }
-}
 
-/// A type enum representing "serialized" GeoArrow geometry types.
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub enum SerializedType {
     /// Represents a [WKBArray][crate::array::WKBArray] or
     /// [ChunkedWKBArray][crate::chunked_array::ChunkedWKBArray] with `i32` offsets.
     WKB(WkbType),
```

Comment:
Since `WkbType` doesn't currently hold the type of binary array, we have `WKB` and `LargeWKB` enum variants (and similar for WKT).

Perhaps it would be better to have a single `WKB` variant and move the data type information into the `WkbType`? Especially so if we expect more array types in the future (e.g. binary view and string view)

But this seems at odds with `ExtensionType::supports_data_type`? Seems like you'd have _one_ extension type that supports a variety of `DataType`, but then it seems weird for the `ExtensionType` to hold a `DataType` inside it...

Or maybe that means here the `Wkb()` variant would hold `Wkb(WkbType, bool)` or `Wkb(WkbType, DataType)`